fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -1298,8 +1343,17 @@ public boolean equals(Object o) {
         }
 
         @Override
-        public int planHash() {
-            return PlanHashable.objectsPlanHash(type, getComparand(), tokenizerName, fallbackTokenizerName);
+        public int planHash(@Nonnull final PlanHashKind hashKind) {
+            switch (hashKind) {
+                case LEGACY:
+                    return PlanHashable.objectsPlanHash(hashKind, type, getComparand(), tokenizerName, fallbackTokenizerName);
```

Comment:
```suggestion
                    return PlanHashable.objectsPlanHash(hashKind, type.name(), getComparand(), tokenizerName, fallbackTokenizerName);
```

Reply:
Done

Reply:
This still doesn't seem to be done. It's strictly speaking an incompatible change, but there is a problem with the `enum` not having a stable hash across instances.

Reply:
Actually, looking at it more carefully, `objectsPlanHash` handled `Enum` properly, so passing `type` and `type.name()` are equivalent, but functionally and in the returned value. We should aim for consistency, I suppose.

Reply:
Will change to use type directly in calls to objectsPlanHash(), across the board.